projects/step-frontend/src/lib/modules/timeseries/dashboard/time-series-dashboard.component.ts

```diff
@@ -0,0 +1,135 @@
+import { AfterViewInit, Component, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { TsFilterItem } from '../performance-view/filter-bar/model/ts-filter-item';
+import { TimeRangePickerSelection } from '../time-selection/time-range-picker-selection';
+import { TimeSeriesContext } from '../time-series-context';
+import { RelativeTimeSelection } from '../time-selection/model/relative-time-selection';
+import { TimeSeriesContextsFactory } from '../time-series-contexts-factory.service';
+import { TimeSeriesDashboardSettings } from './model/ts-dashboard-settings';
+import { PerformanceViewSettings } from '../performance-view/model/performance-view-settings';
+import { TSTimeRange } from '../chart/model/ts-time-range';
+import { PerformanceViewComponent } from '../performance-view/performance-view.component';
+import { RangeSelectionType } from '../time-selection/model/range-selection-type';
+import { filter, Observable, of, Subject, Subscription, takeUntil, combineLatest, throttle, merge, skip } from 'rxjs';
+import { TimeSeriesContextParams } from '../time-series-context-params';
+
+@Component({
+  selector: 'step-timeseries-dashboard',
+  templateUrl: './time-series-dashboard.component.html',
+  styleUrls: ['./time-series-dashboard.component.scss'],
+})
+export class TimeSeriesDashboardComponent implements OnInit, OnDestroy, AfterViewInit {
+  readonly ONE_HOUR_MS = 3600 * 1000;
+
+  @Input() settings!: TimeSeriesDashboardSettings;
+  @ViewChild(PerformanceViewComponent) performanceView!: PerformanceViewComponent;
+
+  timeRangeSelection: TimeRangePickerSelection = { type: RangeSelectionType.FULL };
+  context!: TimeSeriesContext;
+  performanceViewSettings: PerformanceViewSettings | undefined;
+  updateSubscription = new Subscription();
+  update$: Observable<any> | undefined;
```

Comment:
any should be used as a fallback if no other type could be found, please double-check if that is the case